test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -149,15 +149,13 @@ public class AddressRegistrationTests
                 host.Start();
 
                 Assert.Equal(5000, host.GetPort());
-
-                var debugLog = testLogger.Messages.Single(log => log.LogLevel == LogLevel.Debug &&
-                    !log.Message.StartsWith("Hosting start", StringComparison.Ordinal)); // Filter out "Hosting starting" and "Hosting started"
-                Assert.True(string.Equals(debugLog.Message, $"No listening endpoints were configured. Binding to {Constants.DefaultIPEndPoint} by default.", StringComparison.Ordinal));
+                Assert.Equal($"No listening endpoints were configured. Binding to {Constants.DefaultIPEndPoint} by default.",
```

Comment:
Even simpler (sorry, should have realized this earlier):

```csharp
Assert.NotNull(testLogger.Messages.SingleOrDefault(log =>
    log.LogLevel == LogLevel.Debug &&
    log.Message.Equals($"No listening endpoints were configured. Binding to {Constants.DefaultIPEndPoint} by default.", StringComparison.Ordinal)));
```

You get your `StringComparison.Orginal` back :stuck_out_tongue_closed_eyes: